lib/plugins/processor/post.js

```diff
@@ -164,8 +164,10 @@ module.exports = ctx => {
       const id = join(assetDir, item).substring(baseDirLength).replace(/\\/g, '/');
       const asset = PostAsset.findById(id);
 
-      if (asset) return post.published === false ? asset.remove() : undefined; // delete if already exist
-      else if (post.published === false) return undefined; // skip assets for unpulished posts and
+      if (!ctx._showDrafts()) {
```

Comment:
Please make sure to return undefined when the asset exists.
```
if (asset) return undefined;
```

During the running of hexo s, if the source file is modified, this code will still be executed. At this time, the result of PostAsset.findById(id) is not undefined

Reply:
I thought that `asset.remove()` doesn't return anything (= undefined), but I have rewritten the code to return undefined explicitly.

Reply:
Maybe what I said is not clear 😂 

```js
const asset = PostAsset.findById(id);
```

The result of this line of code has two cases, one with a value, and one with `undefined`. When there is a value, we need to return `undefined` in order to skip the following `PostAsset.save()`

But in your modification, if `ctx._showDrafts()` is true, then no matter what the result of `PostAsset.findById(id)` is,  `PostAsset.save()` is executed and not skipped.

Reply:
> I thought that `asset.remove()` doesn't return anything (= undefined), but I have rewritten the code to return undefined explicitly.

Yesh, I also think the code here should be rewritten, GJ

Reply:
@jiangtj Thanks, I think I finally got it 😄

I've refactored the code and commented all the code routes.